### PR TITLE
Fixes various 404 page bugs

### PR DESCRIPTION
**Pull request by itsjoekent**
Also solves #3403 

cc: @aaronschachter

### CODE REVIEW

File: lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module [hunk 1]
```diff
@@ -112,7 +119,7 @@ function dosomething_notfound_preprocess_node(&$vars)  {
 
   $query_string = "";
   $total = count($formatted_terms);
-  for ($i = 1; $i < $total; $i++) { // Cannot start at 0, don't want that part of URL
+  for ($i = 0; $i < $total; $i++) { // Cannot start at 0, don't want that part of URL
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module]: Picky, but lets remove this comment since its not applicable anymore.

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module]: Well I feel like a derp

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_notfound/dosomething_notfound.module]: TOUGHEN UP :muscle:

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**